types/sinon/ts3.1/sinon-tests.ts

```diff
@@ -356,7 +355,12 @@ function testSpy() {
     arr = spy.exceptions;
     arr = spy.returnValues;
 
-    spy('a', 'b');
+    const fnSpy = sinon.spy(fn); // $ExpectType SinonSpy<[string, number], boolean>
+    fn = fnSpy; // Should be assignable to original function
```

Comment:
even if it wasn't assignable to the original, wouldn't typescript just infer a union type here? maybe you should declare what `fn` is above (`let fn: thisbithere = foo`)